### PR TITLE
Update docs and examples

### LINKED ISSUE #494: Architecture specific `update` doesn't work
### Brief description of your issue

I initially wanted to run the normal `update` command:
```sh
wingetcreate update --submit --urls https://todoist.com/windows_app --version 8.12.0 Doist.Todoist
```
But it didn't work:
```log
Each new installer URL must have a single match to an existing installer based on architecture, installer type and scope. The following installers failed to match an existing installer:
Multiple matches found for X86 Nullsoft installer detected from the url: https://todoist.com/windows_app
Try using the architecture and/or scope overrides to uniquely match new URLs to existing installer nodes in the manifest.
```
So I ran the command specifying the new architecture:
```sh
wingetcreate update --submit --urls https://todoist.com/windows_app|x86 --version 8.12.0 Doist.Todoist
```
To which I received an error message:
```log
x86: The term 'x86' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```
Apparently `|` is used as a logical or, but how can I then specify both the URL and the new architecture?

### Steps to reproduce

```sh
wingetcreate update --submit --urls https://todoist.com/windows_app|x86 --version 8.12.0 Doist.Todoist
```

### Expected behavior

The `update` command must create the correct manifest.

### Actual behavior

```log
x86: The term 'x86' is not recognized as a name of a cmdlet, function, script file, or executable program.
Check the spelling of the name, or if a path was included, verify that the path is correct and try again.
```

### Environment

```shell
Windows Package Manager Manifest Creator v1.5.7.0
System Architecture: x64
Windows: 11 v. 23H2
Package: Doist.Todoist 8.12.0
```

### PR DESCRIPTION
**Pull request by mdanish-kh**
- [x] Have you signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs)?
- [x] Are you working against an Issue?
	- Resolves #494
	- Resolves https://github.com/microsoft/winget-create/issues/306
	- Resolves https://github.com/microsoft/winget-create/issues/486
	- Related to https://github.com/microsoft/winget-create/discussions/496#discussioncomment-7967283
	

The commit messages explain specific changes and details

-----

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-create/pull/505)

### CODE REVIEW

File: src/WingetCreateCLI/Commands/UpdateCommand.cs [hunk 1]
```diff
@@ -42,8 +42,8 @@ public static IEnumerable<Example> Examples
...
-                yield return new Example(Resources.Example_UpdateCommand_OverrideArchitecture, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "<InstallerUrl1>|<InstallerArchitecture>" }, Version = "<Version>" });
-                yield return new Example(Resources.Example_UpdateCommand_OverrideScope, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "<InstallerUrl1>|<InstallerScope>" }, Version = "<Version>" });
+                yield return new Example(Resources.Example_UpdateCommand_OverrideArchitecture, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "“<InstallerUrl1>|<InstallerArchitecture>”" }, Version = "<Version>" });
+                yield return new Example(Resources.Example_UpdateCommand_OverrideScope, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "“<InstallerUrl1>|<InstallerScope>”" }, Version = "<Version>" });
```
Comment (mdanish-kh) [inline src/WingetCreateCLI/Commands/UpdateCommand.cs:46]: To display the quotes in the `wingetcreate update --help` helptext, I had to use a unicode characters `“ ”`. The Example class does not display double quotes / single quotes even after escaping them, so I had to use these unicode characters. They are still interpreted correctly in a shell like PowerShell

https://github.com/microsoft/winget-create/assets/88161975/fbb4ea96-68ed-47ca-aa70-f1aed5c4b284

File: src/WingetCreateCLI/Commands/UpdateCommand.cs [hunk 2]
```diff
@@ -42,8 +42,8 @@ public static IEnumerable<Example> Examples
...
                 yield return new Example(Resources.Example_UpdateCommand_SaveAndPublish, new UpdateCommand { Id = "<PackageIdentifier>", Version = "<Version>", OutputDir = "<OutputDirectory>", GitHubToken = "<GitHubPersonalAccessToken>" });
-                yield return new Example(Resources.Example_UpdateCommand_OverrideArchitecture, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "<InstallerUrl1>|<InstallerArchitecture>" }, Version = "<Version>" });
-                yield return new Example(Resources.Example_UpdateCommand_OverrideScope, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "<InstallerUrl1>|<InstallerScope>" }, Version = "<Version>" });
+                yield return new Example(Resources.Example_UpdateCommand_OverrideArchitecture, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "“<InstallerUrl1>|<InstallerArchitecture>”" }, Version = "<Version>" });
```
Comment (ryfu-msft) [inline src/WingetCreateCLI/Commands/UpdateCommand.cs:45]:
```suggestion
                yield return new Example(Resources.Example_UpdateCommand_OverrideArchitecture, new UpdateCommand { Id = "<PackageIdentifier>", InstallerUrls = new string[] { "'<InstallerUrl1>|<InstallerArchitecture>'" }, Version = "<Version>" });

```

I think we can get by with using single quote. I'm sure using a unicode character is fine but probably better to stick with ASCII for now just to be safe. Don't forget to update the docs to reflect this change.

File: doc/update.md [hunk 3]
```diff
@@ -18,6 +18,22 @@ The **update** command can be called with the installer URL(s) that you wish to
 
 There are cases where the intended architecture specified in the existing manifest can sometimes differ from the actual architecture of the installer package. To mitigate this discrepancy, the installer architecture will first be determined by performing a regex string match to identify the possible architecture in the installer url. If no match is found, [Winget-Create](../README.md) will resort to obtaining the architecture from the downloaded installer.
 
+In case Winget-Create fails to obtain the architecture from the binary or the obtained architecture does not match an architecture in the existing manifest, Winget-Create will fail to generate the manifest. In this case, you can explicitly provide the intended architecture in the update command using the following override format:
```
Comment (ryfu-msft) [inline doc/update.md:21]:
```suggestion
If Winget-Create fails to detect the architecture from the binary or the detected architecture does not match an architecture in the existing manifest, Winget-Create will fail to generate the manifest. In this case, you can explicitly provide the intended architecture in the update command using the following override format:
```

**Review by ryfu-msft — CHANGES_REQUESTED** [review]
Thanks for updating the docs! Just a couple suggestions but the rest looks good.

File: src/WingetCreateCLI/Commands/UpdateCommand.cs (hunk 2 above)
Comment (mdanish-kh) [reply src/WingetCreateCLI/Commands/UpdateCommand.cs:45]: That was the first thing I tried.....but....I could swear it wasn't working last time i tried it 😴 smh, how this works perfectly now is beyond me

**Review by ryfu-msft — APPROVED** [review]

**Pull request merged**